src/kernel/src/arch/aarch64/thread.rs

```diff
@@ -81,6 +82,26 @@ where
     todo!()
 }
 
+// The alignment of addresses use by the stack
+const CHECKED_STACK_ALIGNMENT: usize = 16;
+
+/// Compute the top of the stack. 
+/// 
+/// # Safety
+/// The range from [stack_base, stack_base+stack_size] must be valid addresses.
+pub fn new_stack_top(stack_base: usize, stack_size: usize) -> VirtAddr {
+    let stack_addr = (stack_base + stack_size) as u64;
+    // the stack pointer for aarch64 must be aligned to 16 bytes
+    // since the stack is downwards descending, we align the address
+    // down to be within the bounds.
+    let stack_from_args = VirtAddr::new(stack_addr).unwrap();
+    if stack_from_args.is_aligned_to(CHECKED_STACK_ALIGNMENT) {
```

Comment:
nit: I think this if isn't needed, and you can just unconditionally call align_down? up to you

Reply:
I think that would alright. A slightly smaller stack shouldn't cause any issues.

Reply:
I will implement this change in a future PR.